docs: v13 video and TR migration info (#5383)
cypress-io/cypress-documentation: docs/guides/references/migration-guide.mdx

This guide details the changes and how to change your code to migrate to Cypress
version 13.0.
[See the full changelog for version 13.0](https://docs.cypress.io/guides/overview/why-cypress).

### Cypress Cloud Test Replay

[Test Replay](https://docs.cypress.io/guides/overview/why-cypress) is enabled by default in `v13.0` of the Cypress App.

You may need to allowlist `capture.cypress.io` if you work with a strict VPN. See our FAQ section about [VPN subdomain allowlisting](/faq/questions/cloud-faq#Im-working-with-a-restrictive-VPN-Which-subdomains-do-I-have-to-allow-on-my-VPN-for-Cypress-Cloud-to-work-properly).

You can [opt out](https://docs.cypress.io/guides/overview/why-cypress) of this feature in Cloud project-level settings.

<!---

> todo: replace urls for merge to v13 branch:

Reviewer: ```suggestion
TODO: replace urls for merge to v13 branch:
```

changelog- /guides/references/changelog#13-0-0
test replay- /guides/cloud/debugging/test-replay
opt out- /guides/cloud/debugging/test-replay#Opt-out-of-Test-Replay
--->

### Video updates

> #### `video` is set to false by default

Reviewer: ```suggestion
#### `video` is set to `false` by default
```

You can continue recording video, by passing `video: true`, if you want video locally or you want video for some other reason, like in non-Chromium browsers where [Test Replay](https://docs.cypress.io/guides/overview/why-cypress) is not available.

:::cypress-config-example

```ts
{
  video: true
}
```

:::

#### `videoUploadOnPasses` configuration option has been removed

Most users used `videoUploadOnPasses` as a way to skip the time to compress and upload videos to the Cloud. Since we're turning off `videoCompression` by default, this configuration option does not offer the time saving value that it once would.

If you want to prevent a passing test from uploading to the Cloud, we recommend deleting the video using our [guide with code examples to discard captured video of passing tests](/guides/guides/screenshots-and-videos#Control-which-videos-to-keep-and-upload-to-Cypress-Cloud).

> #### `videoCompression` is set to false by default

Reviewer: ```suggestion
#### `videoCompression` is set to `false` by default
```

> Cypress has the capability to compress recorded videos after a run to reduce the video file size. By default, compression is turned off, which results in a larger file size and better video quality.

Reviewer 2: ```suggestion
Cypress has the capability to compress recorded videos after a run to reduce the video file size. By default, compression is now turned off. This results in a reduced run time by removing the time to compress the video, a larger video file size and better video quality.
```

This guide details the changes and how to change your code to migrate to Cypress
version 13.0.
[See the full changelog for version 13.0](https://docs.cypress.io/guides/overview/why-cypress).

### Cypress Cloud Test Replay

[Test Replay](https://docs.cypress.io/guides/overview/why-cypress) is enabled by default in `v13.0` of the Cypress App.

> You may need to whitelist `capture.cypress.io` if you work with a strict VPN. See our FAQ section about [VPN subdomain whitelisting](/faq/questions/cloud-faq#Im-working-with-a-restrictive-VPN-Which-subdomains-do-I-have-to-allow-on-my-VPN-for-Cypress-Cloud-to-work-properly).

Reviewer 3: ```suggestion
You may need to allowlist `capture.cypress.io` if you work with a strict VPN. See our FAQ section about [VPN subdomain allowlisting](/faq/questions/cloud-faq#Im-working-with-a-restrictive-VPN-Which-subdomains-do-I-have-to-allow-on-my-VPN-for-Cypress-Cloud-to-work-properly).
```

See https://www.linkedin.com/pulse/allowlist-blocklist-better-terms-everyone-lets-use-them-rob-black/

Author: I very much appreciate the context of the verbiage change, but I hesitate to introduce a term that isn't widely adopted across the industry.

Reviewer 3: See also GitHub Style Guide:

https://docs.github.com/en/contributing/writing-for-github-docs/style-guide#inclusive-language

I can't say how widely adopted this guidance has become.

Author: TIL! 🧠

You can [opt out](https://docs.cypress.io/guides/overview/why-cypress) of this feature in Cloud project-level settings.

<!---
todo: replace urls for merge to v13 branch:

changelog- /guides/references/changelog#13-0-0
test replay- /guides/cloud/debugging/test-replay
opt out- /guides/cloud/debugging/test-replay#Opt-out-of-Test-Replay
--->

### Video updates

#### `video` is set to false by default

> Whether Cypress will capture a video of the tests run with `cypress run`.

Reviewer 2: I don't understand the point of this line.
```suggestion
```

Author: @jennifer-shehane Yeah, I used the first line under these headings to provide context for what the particular config is. I suppose you may know already if you're here. It does seem a little out of place without more surrounding it. Do you see value in that if it's made more clear in the verbiage or just better without?

You can continue recording video, by passing `video: true`, if you want video locally or you want video for some other reason, like in non-Chromium browsers where [Test Replay](https://docs.cypress.io/guides/overview/why-cypress) is not available.

:::cypress-config-example

```ts
{
  video: true
}
```

:::

#### `videoUploadOnPasses` configuration option has been removed

> Whether Cypress will process, compress, and upload videos to Cypress Cloud even when all tests in a spec file are passing.

Reviewer 2: Again, seems out of place. This should just be clear instructions on how to migrate. 

```suggestion
```

> See our [Control which videos to keep and upload to Cypress Cloud](/guides/guides/screenshots-and-videos#Control-which-videos-to-keep-and-upload-to-Cypress-Cloud) guide with code examples to discard captured video of passing tests.

Reviewer 2: ```suggestion
Most users used `videoUploadOnPasses` as a way to skip the time to compress and upload videos to the Cloud. Since we're turning off `videoCompression` by default, this configuration option does not offer the time saving value that it once would. 

If you want to prevent a passing test from uploading to the Cloud, we recommend deleting the video using our [guide with code examples to discard captured video of passing tests](/guides/guides/screenshots-and-videos#Control-which-videos-to-keep-and-upload-to-Cypress-Cloud).
```

#### `videoCompression` is set to false by default

> The quality setting for the video compression.

Reviewer 2: ```suggestion
```
